### PR TITLE
feature: create api key route

### PR DESCRIPTION
**Pull request by drew-harris**

### CODE REVIEW

File: actix-server/src/handlers/api_key_handler.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+pub struct CreateApiKeyReqPayload {
+    /// The name which will be assigned to the new api key.
+    name: String,
+}
+
+#[derive(Serialize, Deserialize, ToSchema)]
+pub struct CreateApiKeyRespPayload {
+    /// The api key which was created. This is the value which should be used in the Authorization header.
+    api_key: String,
+}
```
Comment (densumesh) [inline actix-server/src/handlers/api_key_handler.rs:17]: Can we change these type names so that they are more distinct

**Review by densumesh — DISMISSED** [review]
LGTM

File: actix-server/src/handlers/api_key_handler.rs (hunk 1 above)
Comment (skeptrunedev) [reply actix-server/src/handlers/api_key_handler.rs:17]: Yeah, I'm game. What are you thinking?

**Comment by drew-harris** [comment]
```
Welcome to the Actix Template CLI! Let's get started by configuring your API Key.
> Would you like to use the default URL for the Actix Template server (http://localhost:8090)? Yes
> Press Enter to authenticate in browser:
> Enter a name for this profile: test-profile
Configuration complete! Your profile has been saved as 'test-profile'.
```
Successfully reaches this point in the "auth" subcommand

**Review by skeptrunedev — APPROVED** [review]

**Pull request merged**